src/VirtoCommerce.Pages.Data/Search/PageDocumentSlugResolver.cs

```diff
@@ -0,0 +1,41 @@
+using System.Linq;
+using System.Threading.Tasks;
+using VirtoCommerce.CoreModule.Core.Seo;
+using VirtoCommerce.Pages.Core.Models;
+using VirtoCommerce.Pages.Core.Search;
+
+namespace VirtoCommerce.Pages.Data.Search;
+
+public class PageDocumentSlugResolver(IPageDocumentSearchService searchService) : ISeoBySlugResolver
+{
+    public async Task<SeoInfo[]> FindSeoBySlugAsync(string slug)
+    {
+        if (string.IsNullOrEmpty(slug))
+        {
+            return [];
+        }
+        if (!slug.StartsWith('/'))
+        {
+            slug = "/" + slug;
+        }
+
+        var criteria = new PageDocumentSearchCriteria
+        {
+            Permalink = slug,
+            Skip = 0,
+            Take = 20
```

Comment:
1. Take 20 Magic
2. Extra Conditions for: Published, Public or Private, Current Date, UserGroups   
3. ObjectType = "Page" ???